### DISCUSSION TITLE
max frequency

### DISCUSSION THREAD

**Discussion by arlimbad222** [fra589/grbl-Mega-5X#250] [Q&A] [answered]
hello @fra589 

i have one simple question on my mind today and that is how fast I can go with the Arduino mega 2560 I share my $$ here can you please guild me what is the best setting for my machine because I found some random crash with these settings I am using original Arduino mega 2560.

$100=100
$101=100
$102=100
$103=8.9

$110=10000
$111=10000
$112=10000
$113=40000

$120=350
$121=350
$122=300
$123=3000

I just wanted to know am I pushing Arduino to its limit???

thank you for your support
[test.txt](https://github.com/fra589/grbl-Mega-5X/files/8027183/test.txt)
.

**Comment by fra589** [comment, marked as answer]
Hi @arlimbad222,  
  
The ATmega2560 have same speed than the chip ATmega328P whicj is on the Arduino Uno.
You can read a discussion about maximum step speed of it here: [https://github.com/grbl/grbl/issues/41](https://github.com/grbl/grbl/issues/41#issuecomment-2412046)
This a discussion concern only the electronic ability. Some additions in the Grbl code since this discussion probably reduced the maximum value which was around 30KHz with 3 axes machine. This may be less with more axes number.  
With my experience, we also need to consider the mechanics limits (friction, moving mass, accelerations, etc.). On my real machine, the maximum speed obtained use a frequency of 5KHz (trapezoidal screws with 2mm pitch, motors Nema 17 with 200 steps per revolution & 1/4 micro-stepping).

See default.h lines 36 to 56 for calculate.

@++;  
Gauthier.
